src/models/DefaultData.js

```diff
@@ -153,51 +109,27 @@ class DefaultData {
    * @return {Boolean} Indicate if component is removed or not.
    */
   removeComponentById(id) {
-    const isRemoved = this.__removeComponentById(this.components, id);
-
-    this.__removeRefAttributeById(this.components, id);
-
-    return isRemoved;
-  }
+    let isRemoved = false;
 
-  /**
-   * Remove components by id.
-   * If not present in the provided components, look on their children.
-   * @param {Component[]} components - Components to search the id.
-   * @param {String} id - Id to remove.
-   * @return {Boolean} Indicate if the component is removed.
-   * @private
-   */
-  __removeComponentById(components, id) {
-    // Remove sonar code smell, because its more optimized with a for loop.
-    for (let index = 0; index < components.length; index += 1) { // NOSONAR
-      if (components[index].id === id) {
-        components.splice(index, 1);
+    this.getChildren(id).forEach((component) => {
+      this.removeComponentById(component.id);
+      isRemoved = true;
+    });
 
+    this.components = this.components.filter((component) => {
+      if (component.id !== id) {
         return true;
       }
-      if (this.__removeComponentById(components[index].children, id)) {
-        return true;
-      }
-    }
+      isRemoved = true;
 
-    return false;
-  }
-
-  /**
-   * Remove id in all attributes' value then if value is empty remove attribute.
-   * @param {Component[]} components - Components to search the id.
-   * @param {String} id - Id to remove.
-   * @private
-   */
-  __removeRefAttributeById(components, id) {
-    components.forEach((component) => {
-      if (component.children.length > 0) {
-        this.__removeRefAttributeById(component.children, id);
-      }
+      return false;
+    });
 
+    this.components.forEach((component) => {
       component.removeLinkAttribute(id);
```

Comment:
will be called even if the component was not removed for some reason. Possible unwanted side effects?

Reply:
No, because the component is not remove only if the id doesn't exists, but we can call this only if file is remove :)